### PR TITLE
feat: Change suggestion components to bind $attrs to base suggestions

### PR DESCRIPTION
**Pull request by annacv**
[EX-7356](https://searchbroker.atlassian.net/browse/EX-7356)

## Motivation and context
Reduce repeating code in components that use `BaseSuggestions` to make them easier to maintain.

**Thgs done**
- Add v-bind=”$attrs” to components that use `BaseSuggestions` & delete props that are not specific to the components itselves
- Link the documentation to the `BaseSuggestions` one, which will have all the `attrs `documented.
- Bind the complete slot props to avoid having to modify the `props` when a new one is added in `BaseSuggestions`

**To take into account:**
As `docgen` is a static analysis tool, it won't resolve variables contained in spread operators. That's why in the documentation parts of code I ended using `v-bind` as the name of the @binding to document. I used `DirectiveFunction` as its type after checking some [documentation](https://github.com/vuejs/vue/blob/main/types/options.d.ts).

The same occurs with props, `docgen` won't resolve the bindings itself, so the props table won't include the props that are not contained in a Vue file (so the table won't be created if there aren't any). This has been solved by creating an "Inherited props" section in the `<docs>` of each file. The con of this strategy is that `eslint-disable max-len` is not working properly inside `mdx` docs section (that's why Jenkins is warning). If checked some [discussions](https://github.com/mdx-js/mdx/discussions/2036) but I'm not sure what should be done in this situation.

A [spike task ](https://searchbroker.atlassian.net/browse/EX-7375?atlOrigin=eyJpIjoiNTA2ODI0YjlhNDhiNDA0YWFmODhkYmEzM2Y3MzU4Y2IiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)is opened regarding this approach as a better solution than the one I explored and see if it also resolves the inherited `props`' issue.

**To test**
Generate documentation locally and check it

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality not to work as expected)
- [x] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?
Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue [hunk 1]
```diff
@@ -78,6 +63,16 @@
...
 
 <docs lang="mdx">
+## Inherited props
+
+This component inherits the [`BaseSuggestions`](../base-components/x-components.base-suggestions.md)
+props.
+
+| Name                          | Description                                                       | Type                | Default       |
+| ----------------------------- | ----------------------------------------------------------------- | ------------------- | ------------- |
+| <code>animation</code>        | Animation component that will be used to animate the suggestions. | <code>Vue</code>    | <code></code> |
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue:73]:
```suggestion
| <code>animation</code>        | Animation component that will be used to animate the suggestions. | <code>Vue</code>    | <code>ul</code> |
```

File: packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue [hunk 2]
```diff
@@ -60,26 +59,20 @@
...
 
 <docs lang="mdx">
+## Inherited props
+
+This component inherits the [`BaseSuggestions`](../base-components/x-components.base-suggestions.md)
+props.
+
+| Name                          | Description                                 | Type                | Default       |
+| ----------------------------- | ------------------------------------------- | ------------------- | ------------- |
+| <code>animation</code>        | Animation component for `QuerySuggestions`. | <code>Vue</code>    | <code></code> |
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue:73]:
```suggestion
| <code>animation</code>        | Animation component for `QuerySuggestions`. | <code>Vue</code>    | <code>ul</code> |
```

File: packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue [hunk 3]
```diff
@@ -1,29 +1,28 @@
...
-        <QuerySuggestion :suggestion="suggestion" class="x-query-suggestions__suggestion">
+      <!-- eslint-enable max-len -->
+      <slot name="suggestion" v-bind="{ ...props }">
+        <QuerySuggestion :suggestion="props.suggestion" class="x-query-suggestions__suggestion">
           <template #default="{ queryHTML }">
             <!-- eslint-disable max-len -->
             <!--
               @slot Custom content that replaces the `QuerySuggestion` default content
-                  @binding {Suggestion} suggestion - Query Suggestion data
+                  @binding {DirectiveFunction} v-bind - Query Suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Query Suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Query Suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue:21]:
```suggestion
                  @binding {Object} v-bind - Query Suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Query Suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Query Suggestion index
```

File: packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue [hunk 4]
```diff
@@ -1,29 +1,28 @@
...
-    :maxItemsToRender="maxItemsToRender"
   >
-    <template #default="{ suggestion, index }">
+    <template #default="props">
+      <!-- eslint-disable max-len -->
       <!--
         @slot Custom component that replaces the `QuerySuggestion` component
-            @binding {Suggestion} suggestion - Query Suggestion data
-            @binding {number} index - Query Suggestion index
+            @binding {DirectiveFunction} v-bind - Query Suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Query Suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Query Suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue:12]:
```suggestion
            @binding {Object} v-bind - Query Suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Query Suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Query Suggestion index
```

File: packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue [hunk 5]
```diff
@@ -41,8 +42,8 @@
   /**
    * Simple popular-searches component that renders a list of suggestions, allowing the user to
    * select one of them, and emitting the needed events.
-   * A popular search is just a query that has been searched a lot in a certain period and may
-   * optionally have associated a set of filters.
+   * A popular search is just a query that has been
+   * searched a lot in a certain period and may optionally have associated a set of filters.
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue:46]: Revert change

```suggestion
   * A popular search is just a query that has been searched a lot in a certain period and may 
   * optionally have associated a set of filters.
```

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue [hunk 6]
```diff
@@ -96,6 +76,16 @@
...
 
 <docs lang="mdx">
+## Inherited props
+
+This component inherits the [`BaseSuggestions`](../base-components/x-components.base-suggestions.md)
+props.
+
+| Name                          | Description                                                                                                                                                                                                  | Type                | Default       |
+| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------- | ------------- |
+| <code>animation</code>        | Animation component that will be used to animate the suggestions.                                                                                                                                            | <code>Vue</code>    | <code></code> |
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/history-queries/components/history-queries.vue:86]:
```suggestion
| <code>animation</code>        | Animation component that will be used to animate the suggestions.                                                                                                                                            | <code>Vue</code>    | <code>ul</code> |
```

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue [hunk 7]
```diff
@@ -55,12 +55,8 @@
 
   /**
    * This component renders a list of suggestions coming from the user queries history.
-   *
-   * @remarks
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/history-queries/components/history-queries.vue:59]: Why did you change this? Just asking out of curiosity 😅

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue [hunk 8]
```diff
@@ -1,41 +1,41 @@
...
+          :suggestion="props.suggestion"
           data-test="history-query-item"
           class="x-history-queries__item"
         >
           <template #default="{ queryHTML }">
             <!-- eslint-disable max-len -->
             <!--
               @slot History Query content
-                  @binding {Suggestion} suggestion - History Query suggestion data
+                  @binding {DirectiveFunction} v-bind - History Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - History Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - History Query suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/history-queries/components/history-queries.vue:25]:
```suggestion
                  @binding {Object} v-bind - History Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - History Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - History Query suggestion index
```

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue [hunk 9]
```diff
@@ -1,41 +1,41 @@
...
+            <slot name="suggestion-content" v-bind="{ ...props, queryHTML }" />
           </template>
-          <template #remove-button-content>
+          <template #remove-button-content="{ ...props }">
+            <!-- eslint-disable max-len -->
             <!--
               @slot History Query remove button content
-                  @binding {Suggestion} suggestion - History Query suggestion data
-                  @binding {number} index - History Query suggestion index
+                  @binding {DirectiveFunction} v-bind - History Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - History Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - History Query suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/history-queries/components/history-queries.vue:35]:
```suggestion
                  @binding {Object} v-bind - History Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - History Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - History Query suggestion index
```

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue [hunk 10]
```diff
@@ -1,41 +1,41 @@
...
-    :maxItemsToRender="maxItemsToRender"
   >
-    <template #default="{ suggestion, index }">
+    <template #default="props">
+      <!-- eslint-disable max-len -->
       <!--
         @slot History Query item
-            @binding {Suggestion} suggestion - History Query suggestion data
-            @binding {number} index - History Query suggestion index
+            @binding {DirectiveFunction} v-bind - History Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - History Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - History Query suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/history-queries/components/history-queries.vue:12]:
```suggestion
            @binding {Object} v-bind - History Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - History Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - History Query suggestion index
```

File: packages/x-components/src/x-modules/next-queries/components/next-queries.vue [hunk 11]
```diff
@@ -1,33 +1,33 @@
...
-    :maxItemsToRender="maxItemsToRender"
   >
-    <template #default="{ suggestion, index }">
+    <template #default="props">
+      <!-- eslint-disable max-len -->
       <!--
         @slot Next Query item
-            @binding {Suggestion} suggestion - Next Query suggestion data
-            @binding {number} index - Next Query suggestion index
+            @binding {DirectiveFunction} v-bind - Next Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Next Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Next Query suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/next-queries/components/next-queries.vue:12]:
```suggestion
            @binding {Object} v-bind - Next Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Next Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Next Query suggestion index
```

File: packages/x-components/src/x-modules/next-queries/components/next-queries.vue [hunk 12]
```diff
@@ -1,33 +1,33 @@
...
           class="x-next-queries__suggestion"
         >
+          <!-- eslint-disable max-len -->
           <!--
               @slot Next Query content
-                  @binding {Suggestion} suggestion - Next Query suggestion data
-                  @binding {boolean} shouldHighlightCurated - True if the curated NQ should
-                  be highlighted
-                  @binding {number} index - Next Query suggestion index
+                  @binding {DirectiveFunction} v-bind - Next Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Next Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Next Query suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/next-queries/components/next-queries.vue:26]:
```suggestion
                  @binding {Object} v-bind - Next Query suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Next Query suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Next Query suggestion index
```

File: packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue [hunk 13]
```diff
@@ -1,26 +1,27 @@
...
-    :animation="animation"
   >
-    <template #default="{ suggestion, index }">
+    <template #default="props">
+      <!-- eslint-disable max-len -->
       <!--
         @slot Popular Search item
-            @binding {Suggestion} suggestion - Popular Search suggestion data
-            @binding {number} index - Popular Search suggestion index
+            @binding {DirectiveFunction} v-bind - Popular Search suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Popular Search suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Popular Search suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue:12]:
```suggestion
            @binding {Object} v-bind - Popular Search suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Popular Search suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Popular Search suggestion index
```

File: packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue [hunk 14]
```diff
@@ -1,26 +1,27 @@
...
+      <!-- eslint-enable max-len -->
+      <slot name="suggestion" v-bind="{ ...props }">
+        <PopularSearch :suggestion="props.suggestion" class="x-popular-searches__suggestion">
+          <template #default="{ ...props }">
+            <!-- eslint-disable max-len -->
             <!--
               @slot Popular Search content
-                  @binding {Suggestion} suggestion - Popular Search suggestion data
-                  @binding {number} index - Popular Search suggestion index
+                  @binding {DirectiveFunction} v-bind - Popular Search suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Popular Search suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Popular Search suggestion index
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue:21]:
```suggestion
                  @binding {Object} v-bind - Popular Search suggestion attributes:<br />&nbsp;&nbsp;- **suggestion** <code>Suggestion</code> - Popular Search suggestion data<br />&nbsp;&nbsp;- **index** <code>number</code> - Popular Search suggestion index
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
good job!! 👏🏿

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue [hunk 15]
```diff
@@ -96,6 +76,16 @@
 </script>
 
 <docs lang="mdx">
```
Comment (tiborux) [inline packages/x-components/src/x-modules/history-queries/components/history-queries.vue:78]: This allows you to ignore the max-length inside mdx
```suggestion
<!--eslint-disable max-len -->
<docs lang="mdx">
```

File: packages/x-components/src/x-modules/next-queries/components/next-queries.vue [hunk 16]
```diff
@@ -110,6 +94,16 @@
 </script>
 
 <docs lang="mdx">
```
Comment (tiborux) [inline packages/x-components/src/x-modules/next-queries/components/next-queries.vue:96]: This allows you to ignore the max-length inside mdx
```suggestion
<!--eslint-disable max-len -->
<docs lang="mdx">
```

File: packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue [hunk 17]
```diff
@@ -78,6 +63,16 @@
 </script>
 
 <docs lang="mdx">
```
Comment (tiborux) [inline packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue:65]: This allows you to ignore the max-length inside mdx
```suggestion
<!--eslint-disable max-len -->
<docs lang="mdx">
```

**Review by tiborux — CHANGES_REQUESTED** [review]
gj

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue (hunk 15 above)
Comment (annacv) [reply packages/x-components/src/x-modules/history-queries/components/history-queries.vue:78]: I tried but didn't work 😭 :  I checked this [discussion](https://github.com/mdx-js/mdx/discussions/2036) but I'm not sure what should be done in this situation.

**Review by tiborux — APPROVED** [review]

File: packages/x-components/src/x-modules/history-queries/components/history-queries.vue (hunk 7 above)
Comment (annacv) [reply packages/x-components/src/x-modules/history-queries/components/history-queries.vue:59]: @LuisMartinez15 it didn't get generated as documentation (if you check [the current docs page](https://docs.empathy.co/develop-empathy-platform/ui-reference/components/history-queries/x-components.history-queries.html), you can see it's not there). I checked the other components applying `baseSuggestions` and they didn't use @remarks, so I thought it was also more consistent to leave it the same way.

**Review by LuisMartinez15 — APPROVED** [review]
good job!

**Review by tiborux — APPROVED** [review]
